pyanaconda/payload/dnf/payload.py

```diff
@@ -174,6 +174,9 @@ def set_from_opts(self, opts):
         if opts.multiLib:
             self.data.packages.multiLib = opts.multiLib
 
+        # Reset all the other things now that we have new configuration.
+        self.unsetup()
```

Comment:
I was looking into that a little more and might be better to directly call `self._configure()`, because `unsetup` does other unnecessary steps (like the tear down of the sources).